sdks/go/pkg/beam/core/runtime/xlangx/expansionx/process_test.go

```diff
@@ -0,0 +1,73 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expansionx
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+func TestNewExpansionServiceRunner(t *testing.T) {
+	testPath := "path/to/jar"
+	serviceRunner := NewExpansionServiceRunner(testPath)
+	if serviceRunner.jarPath != testPath {
+		t.Errorf("JAR path mismatch: wanted %v, got %v", testPath, serviceRunner.jarPath)
+	}
+	commandString := strings.Join(serviceRunner.serviceCommand.Args, " ")
+	expCommandString := "java -jar " + testPath + " 8097"
```

Comment:
This seems like a bit of a brittle test. It might be better to test that the command string contains "java" and "-jar testPath", because those seem like mandatory parts of the command no matter what port you use, or how you modify the command.

Reply:
Fair point. Expanded the test to check for each part of the string individually.